### PR TITLE
1276 increase code coverage

### PR DESCRIPTION
**Pull request by LetsPoke**
added and updated a lot of tests 

only merge after #1277 and #1279

### CODE REVIEW

File: hms-api/src/service/idea-comment-service.ts [hunk 1]
```diff
@@ -32,12 +30,6 @@ export async function createIdeaComment(
     throw new ReferenceNotFoundError(`User with id: ${userId} not found`);
   }
 
-  if (parentCommentId && !(await parentCommentIdIdeaExists(parentCommentId))) {
-    throw new ReferenceNotFoundError(
-      `Parent comment with id: ${parentCommentId} not found`,
-    );
-  }
```
Comment (verzano) [inline hms-api/src/service/idea-comment-service.ts:39]: why is this removed?

File: hms-api/src/handler/ideaComment/create.ts [hunk 2]
```diff
@@ -7,10 +7,9 @@ import IdeaCommentCreateResponse from '../../rest/ideaComment/IdeaCommentCreateR
 export async function create(event, context, callback) {
   await wrapHandler(async () => {
     const request = IdeaCommentCreateRequest.parse(event.body);
-    const ideaId = event.pathParameters.id;
     const ideaComment = await createIdeaComment(
       request.userId,
-      ideaId,
```
Comment (verzano) [inline hms-api/src/handler/ideaComment/create.ts:13]: idk about this 🤔

File: hms-api/src/error/ValidationError.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import ValidationResult from './ValidationResult';
 
 class ValidationError extends Error {
   constructor(message: string, result: ValidationResult) {
-    super(message + ' ' + result.toBulletList());
```
Comment (verzano) [inline hms-api/src/error/ValidationError.ts:7]: was it making double spaces?

Comment (LetsPoke) [reply hms-api/src/error/ValidationError.ts:7]: yes it always added one extra space to the "message"

File: hms-api/src/service/idea-comment-service.ts (hunk 1 above)
Comment (LetsPoke) [reply hms-api/src/service/idea-comment-service.ts:39]: for consistency as we nowwhere else check for partencommentId neither test for it 
just to have it completly removed for now until we someday add partenCommentIds

File: hms-api/src/handler/ideaComment/create.ts (hunk 2 above)
Comment (LetsPoke) [reply hms-api/src/handler/ideaComment/create.ts:13]: im sure there was a good reason why i did this 
maybe it was just for consistency with all the other create methods

File: hms-api/src/service/idea-comment-service.ts (hunk 1 above)
Comment (verzano) [reply hms-api/src/service/idea-comment-service.ts:39]: then I would argue that you should either leave it (so that you don't have to remember later to put it back) or you remove everything related to parent comments

File: hms-api/src/handler/ideaComment/create.ts (hunk 2 above)
Comment (verzano) [reply hms-api/src/handler/ideaComment/create.ts:13]: well, we SHOULD be pulling it from the path and not from the request

Comment (verzano) [reply hms-api/src/handler/ideaComment/create.ts:13]: or, we should remove it from the path (means changing the frontend too) and only get it from the request

File: hms-api/src/service/idea-comment-service.ts (hunk 1 above)
Comment (LetsPoke) [reply hms-api/src/service/idea-comment-service.ts:39]: just think about as it was never there

File: hms-api/src/handler/ideaComment/create.ts (hunk 2 above)
Comment (LetsPoke) [reply hms-api/src/handler/ideaComment/create.ts:13]: if this dont make the tests impossible then i can try it

File: hms-api/src/service/idea-comment-service.ts (hunk 1 above)
Comment (verzano) [reply hms-api/src/service/idea-comment-service.ts:39]: but, it IS there.  In the function arguments.  There is no reason to not just leave this here
